compiler/rustc_lint/src/levels.rs

```diff
@@ -115,6 +115,38 @@ impl LintLevelSets {
     }
 }
 
+fn lints_that_dont_need_to_run(tcx: TyCtxt<'_>, (): ()) -> FxIndexSet<LintId> {
+    let store = unerased_lint_store(&tcx.sess);
```

Comment:
Future-compatibility lints that show up in cargo's reports are supposed to be emitted even if they are allowed in the crate. Isn't that something that this logic needs to account for somehow? Or do we account for this already in a different way?

Reply:
I opened https://github.com/rust-lang/rust/pull/133108 to fix this.